tracer/src/Datadog.Trace/Logging/DirectSubmission/ImmutableDirectLogSubmissionSettings.cs

```diff
@@ -0,0 +1,221 @@
+﻿// <copyright file="ImmutableDirectLogSubmissionSettings.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging.DirectSubmission.Sink.PeriodicBatching;
+
+namespace Datadog.Trace.Logging.DirectSubmission
+{
+    /// <summary>
+    /// Contains direct-log-submission-specific settings
+    /// </summary>
+    internal class ImmutableDirectLogSubmissionSettings
+    {
+        internal static readonly IntegrationId[] SupportedIntegrations =
+        {
+            IntegrationId.Serilog,
+            IntegrationId.ILogger,
+            IntegrationId.Log4Net,
+            IntegrationId.NLog,
+        };
+
+        private readonly bool[] _enabledIntegrations;
+
+        private ImmutableDirectLogSubmissionSettings(
+            string host,
+            string source,
+            string globalTags,
+            Uri intakeUrl,
+            string apiKey,
+            bool isEnabled,
+            DirectSubmissionLogLevel minimumLevel,
+            bool[] enabledIntegrations,
+            List<string> validationErrors,
+            List<string> enabledIntegrationNames,
+            BatchingSinkOptions batchingOptions)
+        {
+            Host = host;
+            Source = source;
+            GlobalTags = globalTags;
+            IntakeUrl = intakeUrl;
+            ApiKey = apiKey;
+            ValidationErrors = validationErrors;
+            EnabledIntegrationNames = enabledIntegrationNames;
+            MinimumLevel = minimumLevel;
+            _enabledIntegrations = enabledIntegrations;
+            IsEnabled = isEnabled;
+            BatchingOptions = batchingOptions;
+        }
```

Comment:
It's not a huge deal, but why not accept the DirectLogSubmissionSettings to clean this up?

Reply:
It comes down to where we do the verification/correction for invalid values. I could (should?) remove the `Create()` factory method and move it into the constructor instead, but I've always kinda liked this pattern. Meh. I'll leave it for now, can easily be refactored later.